packages/libs/components/src/plots/VolcanoPlot.tsx

```diff
@@ -0,0 +1,274 @@
+  const {
+    data = EmptyVolcanoPlotData,
+    independentAxisRange,
+    dependentAxisRange,
+    significanceThreshold,
+    log2FoldChangeThreshold,
+    markerBodyOpacity,
+    ...restProps
+  } = props;
+
+  /**
+   * Find mins and maxes of the data and for the plot
+   */
+
+  const dataXMin = min(data.foldChange.map(Number));
+  const dataXMax = max(data.foldChange.map(Number));
+  const dataYMin = min(data.pValue.map(Number));
+  const dataYMax = max(data.pValue.map(Number));
+
+  // Determine mins, maxes of axes in the plot.
+  // These are different than the data mins/maxes because
+  // of the log transform and the little bit of padding.
+  //
+
+  let xMin: number;
+  let xMax: number;
+  let yMin: number;
+  let yMax: number;
+
+  // Log transform for plotting, and add a little margin for axes
+  if (dataXMin && dataXMax) {
+    xMin = Math.log2(dataXMin);
+    xMax = Math.log2(dataXMax);
+    xMin = xMin - (xMax - xMin) * 0.05;
+    xMax = xMax + (xMax - xMin) * 0.05;
+  } else {
+    xMin = 0;
+    xMax = 0;
+  }
+  if (dataYMin && dataYMax) {
+    yMin = -Math.log10(dataYMax);
+    yMax = -Math.log10(dataYMin);
+    yMin = yMin - (yMax - yMin) * 0.05;
+    yMax = yMax + (yMax - yMin) * 0.05;
+  } else {
+    yMin = 0;
+    yMax = 0;
+  }
+  console.log(yMin);
+
+  /**
+   * Turn the data (array of arrays) into data points (array of points)
+   */
+
+  let dataPoints: DataPoint[] = [];
+
+  // Loop through the data and return points. Doesn't really matter
+  // which var of the data we map over.
+  data.foldChange.forEach((fc, ind: number) => {
+    dataPoints.push({
+      foldChange: fc,
+      pValue: data.pValue[ind],
+      adjustedPValue: data.adjustedPValue[ind],
+      pointId: data.pointId[ind],
+      color: assignSignificanceColor(
+        Math.log2(Number(fc)),
+        Number(data.pValue[ind]),
+        significanceThreshold,
+        log2FoldChangeThreshold,
+        significanceColors
+      ),
+    });
+  });
+
+  /**
+   * Accessors - tell visx which value of each points we should use and where.
+   */
+
+  const dataAccessors = {
+    xAccessor: (d: any) => {
+      return Math.log2(d?.foldChange);
+    },
+    yAccessor: (d: any) => {
+      return -Math.log10(d?.pValue);
+    },
+  };
+
+  const thresholdLineAccessors = {
+    xAccessor: (d: any) => {
+      return d?.x;
+    },
+    yAccessor: (d: any) => {
+      return d?.y;
+    },
+  };
+
+  /**
+   * Plot styles
+   * (can eventually be moved to a new file and applied as a visx theme)
+   */
+  const thresholdLineStyles = {
+    stroke: '#aaaaaa',
+    strokeWidth: 1,
+    strokeDasharray: 3,
+  };
+  const axisStyles = {
+    stroke: '#bbbbbb',
+    strokeWidth: 1,
+  };
+  const gridStyles = {
+    stroke: '#dddddd',
+    strokeWidth: 0.5,
+  };
+  console.log(yMin);
+
+  return (
+    // From docs " For correct tooltip positioning, it is important to wrap your
+    // component in an element (e.g., div) with relative positioning."
```

Comment:
**random**: If you or anyone would like to understand absolute/relative positioning, [this Codepen might help](https://codepen.io/skendrick366/pen/WNabrep).

The docs say this because it sounds like the tooltips are positioned absolutely. Absolutely position elements position themselves in relation to their nearest parent that's positioned relatively. 🤯 

You likely know this, but I just wanted to put it in there in case!

Reply:
i did not know this! Thank you! That was a really clear example